### PR TITLE
boards: mimxrt1040_evk: add LinkServer and Pyocd

### PR DESCRIPTION
**Pull request by butok**
Add LinkServer and Pyocd runners for the MIMXRT1040-EVK board.

### CODE REVIEW

File: boards/arm/mimxrt1040_evk/board.cmake [hunk 1]
```diff
@@ -5,5 +5,9 @@
 #
 
 board_runner_args(jlink "--device=MIMXRT1042XXX6B")
+board_runner_args(linkserver "--device=MIMXRT1042xxxxB:MIMXRT1040-EVK")
+board_runner_args(pyocd "--target=MIMXRT1042XJM5B")
```
Comment (danieldegrasse) [inline boards/arm/mimxrt1040_evk/board.cmake:9]: This `target` argument isn't working for me with pyocd 0.36.0, has support been added for the RT1040 EVK? `pyocd list -t` does not show it as a target:
image

File: boards/arm/mimxrt1040_evk/board.cmake [hunk 2]
```diff
@@ -5,5 +5,9 @@
 #
 
 board_runner_args(jlink "--device=MIMXRT1042XXX6B")
+board_runner_args(linkserver "--device=MIMXRT1042xxxxB:MIMXRT1040-EVK")
```
Comment (danieldegrasse) [inline boards/arm/mimxrt1040_evk/board.cmake:8]: Verified I was able to flash an RT1040 EVK with the above enablement

File: boards/arm/mimxrt1040_evk/board.cmake (hunk 1 above)
Comment (butok) [reply boards/arm/mimxrt1040_evk/board.cmake:9]: > This `target` argument isn't working for me with pyocd 0.36.0

The PyOCD works with the RT1042 CMSIS-DFP.

File: boards/arm/mimxrt1040_evk/board.cmake (hunk 2 above)
Comment (butok) [reply boards/arm/mimxrt1040_evk/board.cmake:8]: > Verified I was able to flash an RT1040 EVK with the above enablement

Me too ;)

File: boards/arm/mimxrt1040_evk/board.cmake (hunk 1 above)
Comment (danieldegrasse) [reply boards/arm/mimxrt1040_evk/board.cmake:9]:
```
0004336 C Target type mimxrt1042xjm5b not recognized. Use 'pyocd list --targets' to see currently available target types. See <https://pyocd.io/docs/target_support.html> for how to install additional target support. [__main__]
FATAL ERROR: command exited with status 1: pyocd flash -e sector -a 0x60000000 -t MIMXRT1042XJM5B /home/danieldegrasse/zephyr/zephyrproject/zephyr/build/zephyr/zephyr.bin
```

This is what I see locally. What commands are you running to test pyocd? I am running the following:
```
$ west build -p always -b mimxrt1040_evk samples/hello_world
$ west flash --runner=pyocd
```

Comment (butok) [reply boards/arm/mimxrt1040_evk/board.cmake:9]: > ```
> 0004336 C Target type mimxrt1042xjm5b not recognized. Use 'pyocd list --targets' to see currently available target types. See <https://pyocd.io/docs/target_support.html> for how to install additional target support. [__main__]
> FATAL ERROR: command exited with status 1: pyocd flash -e sector -a 0x60000000 -t MIMXRT1042XJM5B /home/danieldegrasse/zephyr/zephyrproject/zephyr/build/zephyr/zephyr.bin
> ```
> 
> This is what I see locally. What commands are you running to test pyocd? I am running the following:
> 
> ```
> $ west build -p always -b mimxrt1040_evk samples/hello_world
> $ west flash --runner=pyocd
> ```

Hi David,
The PyOCD works with the CMSIS-DFP packs. It means it is able to work with any platforms, not only built-in.
More info here: https://pyocd.io/docs/target_support.html
So I have downloaded NXP.MIMXRT1042_DFP.17.0.0.pack, and added pyocd.yaml (to the zephyr root folder)  with the path to the package:
```
pack:
  - C:\_ddm\CMSISPACKS\NXP.MIMXRT1042_DFP.17.0.0.pack
```
After that, PyOCD see the RT1042
```
pyocd list -t
...
 mimxrt1010                NXP                      MIMXRT1011xxxxx                           builtin
  mimxrt1015                NXP                      MIMXRT1015xxxxx                           builtin
  mimxrt1020                NXP                      MIMXRT1021xxxxx                           builtin
  mimxrt1024                NXP                      MIMXRT1024xxxxx                           builtin
  mimxrt1042dfp6b           NXP                      MIMXRT1042DFP6B              MIMXRT1042   pack
  mimxrt1042xfp5b           NXP                      MIMXRT1042XFP5B              MIMXRT1042   pack
  mimxrt1042xjm5b           NXP                      MIMXRT1042XJM5B              MIMXRT1042   pack
  mimxrt1042xxxxb           NXP                      MIMXRT1042xxxxB              MIMXRT1042   pack
  mimxrt1050                NXP                      MIMXRT1052xxxxB_hyperflash                builtin
  mimxrt1050_hyperflash     NXP                      MIMXRT1052xxxxB_hyperflash                builtin
..
```

Comment (danieldegrasse) [reply boards/arm/mimxrt1040_evk/board.cmake:9]: @butok Thank you for the clarification- I wasn't aware pyocd offered this feature. I was able to get support for pyocd working on my end with this DFP

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**